eventrecorder/event.go

```diff
@@ -0,0 +1,124 @@
+package eventrecorder
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	types "github.com/filecoin-project/lassie/pkg/types"
+	"github.com/google/uuid"
+)
+
+var (
+	validPhases     = []string{"indexer", "query", "retrieval"}
+	validEventNames = []string{
+		"accepted",
+		"candidates-filtered",
+		"candidates-found",
+		"connected",
+		"failure",
+		"first-byte-received",
+		"proposed",
+		"query-asked",
+		"query-asked-filtered",
+		"started",
+		"success",
+	}
+)
+
+type event struct {
+	RetrievalId       *types.RetrievalID `json:"retrievalId"`
+	InstanceId        *string            `json:"instanceId,omitempty"`
+	Cid               *string            `json:"cid"`
+	StorageProviderId *string            `json:"storageProviderId"`
+	Phase             *types.Phase       `json:"phase"`
+	PhaseStartTime    *time.Time         `json:"phaseStartTime"`
+	EventName         *types.EventCode   `json:"eventName"`
+	EventTime         *time.Time         `json:"eventTime"`
```

Comment:
Are these all pointers because they are optional fields? Validation implementation suggests otherwise. Which leaves the question why pointers?

Reply:
I needed them to be pointers because I need to check if their values are nil to gaurd against missing properties.

Reply:
I am curious why not check for zero values?